pkg/k8s/apps/translate.go

```diff
@@ -561,44 +568,71 @@ func TranslateOktetoInitFromImageContainer(spec *apiv1.PodSpec, rule *model.Tran
 	spec.InitContainers = append(spec.InitContainers, *c)
 }
 
-// TranslateOktetoSyncSecret translates the syncthing secret container of a pod
-func TranslateOktetoSyncSecret(spec *apiv1.PodSpec, name string) {
-	if spec.Volumes == nil {
-		spec.Volumes = []apiv1.Volume{}
+func isOktetoSyncSecretVolumePresent(spec *apiv1.PodSpec) bool {
+	for _, v := range spec.Volumes {
+		if v.Name == oktetoSyncSecretVolume {
+			return true
+		}
 	}
-	for i := range spec.Volumes {
-		if spec.Volumes[i].Name == oktetoSyncSecretVolume {
-			return
+	return false
+}
+
+func isOktetoSyncDataVolumePresent(spec *apiv1.PodSpec) bool {
+	for _, v := range spec.Volumes {
+		if v.Name == oktetoSyncDataVolume {
+			return true
 		}
 	}
+	return false
+}
 
-	var mode int32 = 0444
-	v := apiv1.Volume{
-		Name: oktetoSyncSecretVolume,
-		VolumeSource: apiv1.VolumeSource{
-			Secret: &apiv1.SecretVolumeSource{
-				SecretName: fmt.Sprintf(oktetoSecretTemplate, name),
-				Items: []apiv1.KeyToPath{
-					{
-						Key:  "config.xml",
-						Path: "config.xml",
-						Mode: &mode,
-					},
-					{
-						Key:  "cert.pem",
-						Path: "cert.pem",
-						Mode: &mode,
-					},
-					{
-						Key:  "key.pem",
-						Path: "key.pem",
-						Mode: &mode,
+// TranslateOktetoSyncthingVolumes translates the syncthing secret container of a pod
+func TranslateOktetoSyncthingVolumes(spec *apiv1.PodSpec, name string) {
+	if spec.Volumes == nil {
+		spec.Volumes = []apiv1.Volume{}
```

Comment:
@jLopezbarb the unit test `Test_translateWithVolumes` needs to be updated reflecting these changes